### DISCUSSION TITLE
Installed two plugins and they appear in the GUI but they fail to run

### DISCUSSION THREAD

**Discussion by andy-butterworth** [netbox-community/netbox#11161] [General]
I have installed the plugins 'netbox-bgp' and 'netbox-acls' and they both appear on the left under plugins.  However selecting any of them results in an 'Internal server error':

```
<class 'django.db.utils.ProgrammingError'>

relation "netbox_bgp_bgppeergroup" does not exist
LINE 1: SELECT (1) AS "a" FROM "netbox_bgp_bgppeergroup" LIMIT 1
                               ^


Python version: 3.8.10
NetBox version: 3.3.9
```

I'm fairly sure this is because I've added the plugins after the original install, but I can't see anything telling me how to fix it.

**Comment by mtinberg** [comment]
This error looks like the DB schema hasn't been created yet. I'm not sure, but you may need to run upgrade.sh (or at least manage.py migrate) so the plugin can create the DB tables specified in its model(s).

Looking at the docs this is the case

https://demo.netbox.dev/static/docs/plugins/

Run db migrations, collect static files, restart WSGI service are all general steps for plugin deployment, that may not be specifically included in any particular plugin's README.md


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: andy-butterworth ***@***.***>
Sent: Monday, December 12, 2022 9:47 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Installed two plugins and they appear in the GUI but they fail to run (Discussion #11161)


I have installed the plugins 'netbox-bgp' and 'netbox-acls' and they both appear on the left under plugins. However selecting any of them results in an 'Internal server error':

<class 'django.db.utils.ProgrammingError'>

relation "netbox_bgp_bgppeergroup" does not exist
LINE 1: SELECT (1) AS "a" FROM "netbox_bgp_bgppeergroup" LIMIT 1
                               ^


Python version: 3.8.10
NetBox version: 3.3.9


I'm fairly sure this is because I've added the plugins after the original install, but I can't see anything telling me how to fix it.

**Reply by andy-butterworth** [reply]
Thanks for the reply Mark.  After some more reading I realised this and have now sorted it.
